app/presenters/document/web_resource_presenter.rb

```diff
@@ -0,0 +1,175 @@
+    def media_type
+      @media_type ||= begin
+        if @record_presenter.iiif_manifesto
+          'iiif'
+        else
+          case (mime_type || '').downcase
+          when /^audio\//
+            'audio'
+          when /^image\//
+            'image'
+          when /^video\//
+            'video'
+          when /^text\//, /\/pdf$/
+            'text'
+          else
+            @record_presenter.render_document_show_field_value('type')
+          end
+        end
+      end
+    end
+
+    def edm_media_type
+      @edm_media_type ||= begin
+        record_type = @record_presenter.render_document_show_field_value('type')
+        if record_type == '3D' || media_type == 'iiif'
+          record_type
+        elsif media_type == 'audio'
+          'SOUND'
+        else
+          media_type.upcase
+        end
+      end
+    end
+
+    def use_media_proxy?
+      Rails.application.config.x.edm_is_shown_by_proxy &&
+        mime_type.present? &&
+        mime_type.match('image/').nil?
+    end
+
+    def download_url
+      @download_url ||= begin
+        if use_media_proxy?
+          Rails.application.config.x.edm_is_shown_by_proxy + @record.fetch('about', '/') + '?view=' + CGI.escape(url)
+        else
+          url
+        end
+      end
+    end
+
+    def technical_metadata
+      file_size = number_to_human_size(render_document_show_field_value('ebucoreFileByteSize')) || ''
+      {
+        mime_type: mime_type,
+        file_size: file_size.split(' ').first,
+        file_unit: file_size.split(' ').last,
+        codec: render_document_show_field_value('edmCodecName'),
+        width: render_document_show_field_value('ebucoreWidth'),
+        height: render_document_show_field_value('ebucoreHeight'),
+        size_unit: 'pixels',
+        runtime: render_document_show_field_value('ebucoreDuration'),
+        runtime_unit: 'seconds'
+      }
+    end
+
+    def playable?
+      if url.blank? ||
+          (media_type != 'iiif' && mime_type.blank?) ||
+          (mime_type == 'video/mpeg') ||
+          (media_type == 'text' && mime_type == 'text/plain; charset=utf-8') ||
+          (media_type == 'video' && mime_type == 'text/plain; charset=utf-8')
+        false
+      else
+        true
+      end
+    end
+
+    def downloadable?
+      if url.blank? ||
+          mime_type.blank? ||
+          download_disabled? ||
+          media_type == 'iiif' ||
+          (media_type == 'text' && mime_type == 'text/plain; charset=utf-8') ||
+          (media_type == 'video' && mime_type == 'text/plain; charset=utf-8')
+        false
+      else
+        true
+      end
+    end
+
+    def download_disabled?
+      blacklisted = %w(http://www.europeana.eu/rights/rr-p/ http://www.europeana.eu/rights/rr-r/)
+      blacklisted.include?(media_rights)
+    end
+
+    def thumbnail
+      Europeana::API.url + '/thumbnail-by-url.json?size=w400&uri=' + CGI.escape(url) + '&type=' + edm_media_type
+    end
+
+    def player
```

Comment:
Cyclomatic complexity for player is too high. [8/6]<br>Method has too many lines. [16/10]